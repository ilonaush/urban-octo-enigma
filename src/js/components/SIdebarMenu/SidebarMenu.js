import React from 'react';
import {MENU_ITEMS} from "../../constants";
import {NavLink} from "react-router-dom";
import "./SidebarMenu.styl";

const SidebarMenu = () => {
  return (
      <ul>
        {MENU_ITEMS.map((link) => {
              return (
                  <li key={link.name}>
                    <NavLink to={link.to}>
                      {link.name}
                    </NavLink>
                  </li>
              )
            }
        )}
      </ul>
  );
};


export default SidebarMenu;
