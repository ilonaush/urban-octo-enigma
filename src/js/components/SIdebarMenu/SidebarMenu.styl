ul
  padding  0
  li
    display block
    padding 5px
    text-align center
    margin: 0 10px

    a
      text-decoration none
      color white
      width 100%
      display flex
      align-items center

    .icon
      height 20px
      width 20px
      display block
      background-repeat no-repeat
      margin-right 30px
      margin-left 10px
