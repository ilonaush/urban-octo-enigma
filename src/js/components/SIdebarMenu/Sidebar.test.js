import React from 'react';

import SidebarMenu  from './SidebarMenu';

import ReactDOM from 'react-dom';
import {MemoryRouter} from "react-router";

describe('SidebarMenu', () => {
    it('renders', () => {

        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter><SidebarMenu/></MemoryRouter>, div)
    })
})

