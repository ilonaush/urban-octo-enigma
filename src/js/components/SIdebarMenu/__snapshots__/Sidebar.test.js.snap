// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu render correctly sidebar component 1`] = `
<div>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          alt=""
          className="icon"
          style={
            Object {
              "backgroundImage": "url('images/list.svg",
            }
          }
        />
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/add-worker"
        onClick={[Function]}
      >
        <span
          alt=""
          className="icon"
          style={
            Object {
              "backgroundImage": "url('images/add.svg",
            }
          }
        />
        Add a worker
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/fire-worker"
        onClick={[Function]}
      >
        <span
          alt=""
          className="icon"
          style={
            Object {
              "backgroundImage": "url('images/fire.svg",
            }
          }
        />
        Fire worker
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/gallery"
        onClick={[Function]}
      >
        <span
          alt=""
          className="icon"
          style={
            Object {
              "backgroundImage": "url('images/gallery.svg",
            }
          }
        />
        Gallery
      </a>
    </li>
  </ul>
</div>
`;
